### ISSUE TITLE
why the results of tf.contrib.crf.viterbi_decode is different from tf.contrib.crf.crf_decode?

### ISSUE THREAD

**Issue by boredbird** [tensorflow/tensorflow#21538]
### Describe the problem
For viterbi decode,i have try the both api:
tf.contrib.crf.viterbi_decode and tf.contrib.crf.crf_decode.
I got different results.

### Source code 
the way i use  tf.contrib.crf.viterbi_decode:
            for logit, sequence_length in zip(logits, sequence_lengths):
                logit = logit[:sequence_length] # keep only the valid steps
                viterbi_seq, viterbi_score = tf.contrib.crf.viterbi_decode(
                        logit, trans_params)
the way i use  tf.contrib.crf.crf_decode:
 viterbi_sequence, viterbi_score = tf.contrib.crf.crf_decode(self.logits, self.trans_params,self.sequence_lengths)

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by tensorflowbutler** [comment]
It has been 20 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 36 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by harshini-gadige** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
